api/coreservice.go

```diff
@@ -1379,25 +1384,45 @@ func (core *coreService) LogsInRange(filter *logfilter.LogFilter, start, end, pa
 		return nil, err
 	}
 
-	// TODO: improve using goroutine
 	if paginationSize == 0 {
 		paginationSize = 1000
 	}
 	if paginationSize > 5000 {
 		paginationSize = 5000
 	}
+
+	c := make(chan _logsInBlock)
 	logs := []*iotextypes.Log{}
+	if len(blockNumbers) == 0 {
+		return logs, nil
+	}
 	for _, i := range blockNumbers {
-		logsInBlock, err := core.LogsInBlock(filter, i)
-		if err != nil {
+		go func(blockNumber uint64) {
+			logsInBlock, err := core.logsInBlock(filter, blockNumber)
+			ret := &_logsInBlock{
+				logsInBlock,
+				err,
+			}
+			c <- *ret
+		}(i)
+	}
+
+	i := 0
+	for logsInBlock := range c {
```

Comment:
Msg from channel might not come in order

Reply:
Still solving this, any suggestion?